impeller/renderer/backend/vulkan/tracked_objects_vk.cc

```diff
@@ -43,28 +43,30 @@ bool TrackedObjectsVK::IsValid() const {
   return is_valid_;
 }
 
-void TrackedObjectsVK::Track(std::shared_ptr<SharedObjectVK> object) {
+void TrackedObjectsVK::Track(const std::shared_ptr<SharedObjectVK>& object) {
   if (!object || (!tracked_objects_.empty() &&
                   object.get() == tracked_objects_.back().get())) {
     return;
   }
-  tracked_objects_.emplace_back(std::move(object));
+  tracked_objects_.emplace_back(object);
```

Comment:
This is incrementing now.  The thesis is that this increment matches the one we are replacing when we grab a const ref and in the case where we leave early we avoid the increment altogether?  SGTM.  I wish we had settings on that required an explicit copy here.

Reply:
Yeah, though I found from tracing that we still have multiple share_ptr decrements per draw because the buffer_view emplacement also does a move still. I will send a separate patch for that